presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -541,6 +543,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            Optional<Expression> argumentNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
```

Comment:
I see that https://github.com/prestodb/presto/pull/7354 got merged. When I rebase I will start work on supporting better column resolution in the background but I don't think it's absolutely necessary for this PR to go through.

As for your second comment, that query fails at the moment (on latest master) without any grouping -- `select 1 as x from tpch.sf1.nation group by x` fails with `Column 'x' cannot be resolved`. Also queries where the arguments in grouping refer to columns being projected work in Postgres. Are you talking about aliasing? So you can't refer to columns in grouping by their alias?